modules/rostests/apitests/browseui/SHExplorerParseCmdLine.c

```diff
@@ -209,16 +209,18 @@ START_TEST(SHExplorerParseCmdLine)
         { __LINE__, L"%wrongdir%", TRUE, PIDL_IS_NULL, 0x02000000, L"%wrongdir%"},
         { __LINE__, L"%programfiles#", TRUE, PIDL_IS_NULL, 0x02000000, L"%programfiles#"},
         { __LINE__, L",", TRUE, PIDL_IS_EMPTY, 0x00000200},
+        { __LINE__, L"\\", TRUE, PIDL_IS_PATH, 0x00000200, NULL, L"C:\\"}, // disk letter depends on current directory
         { __LINE__, L"c:\\", TRUE, PIDL_IS_PATH, 0x00000200, NULL, L"C:\\" },
         { __LINE__, L"c:", TRUE, PIDL_IS_PATH, 0x00000200, NULL, L"C:\\" },
         { __LINE__, L"c", TRUE, PIDL_IS_NULL, 0x02000000, L"c"},
         { __LINE__, L"c:\\Program Files", TRUE, PIDL_IS_PATH, 0x00000200, NULL, L"C:\\Program Files" },
         { __LINE__, L"c:\\Program Files\\", TRUE, PIDL_IS_PATH, 0x00000200, NULL, L"C:\\Program Files" },
         { __LINE__, L"c:\\Program Files/", TRUE, PIDL_IS_NULL, 0x02000000, L"c:\\Program Files/"},
         { __LINE__, L"c:/Program Files/", TRUE, PIDL_IS_NULL, 0x02000000, L"c:/Program Files/"},
-        { __LINE__, L"fonts", TRUE, PIDL_IS_NULL, 0x02000000, L"fonts" },
-        { __LINE__, L"winsxs", TRUE, PIDL_IS_NULL, 0x02000000, L"winsxs" },
-        { __LINE__, L"system32", TRUE, PIDL_IS_NULL, 0x02000000, L"system32" },
+        { __LINE__, L"fonts", TRUE, PIDL_IS_PATH, 0x00000200, NULL, L"C:\\WINDOWS\\fonts" }, // works if test started from C:\WINDOWS\ directory
```

Comment:
It looks like this test is succeeding on WHS testbot, so changing it seems odd.

Can we please fix the test so it doesn't depend on the current directory at all? A simple GetWindowsDirectory/SetCurrentDirectory would be fine, then it wouldn't matter where it was launched from.

Reply:
On that note, are the tests expecting "Documents and Settings\gigaherz" not failing for you? I would expect they need to be fixed to point to C:\ and C:\Windows as well

Reply:
I think the tests that include my nickname don't actually require the path to exist? I recall them passing in reactos and win2003, which have different usernames than "gigaherz" in them... Still, probably best not to use my nickname in the tests for no reason... :P

Reply:
@HBelusca looks like it's not possible to not hardcode variables here, I'm getting error `initializer element is not constant`.

Reply:
Yes, this needs to be solved by doing an RtlStringCbPrintfW on the expected string, replacing all `%`s with `%%` and passing the Windows directory as the format argument so that %ls can be used as a placeholder for the Windows dir.

Reply:
The disk letter can be dynamic too:
https://github.com/reactos/reactos/pull/159/files#diff-f1ad310081d3725e4d93e2b49421d77fR212

I think it would be simpler to solve this with something like string replacements.

Reply:
You just removed a valid test here. Calling ShellExecutExW with "fonts" as parameter succeeds even if the current path in explorer isn't C:\WINDOWS\ but SHExplorerParseCmdLine fails.

Reply:
SHExplorerParseCmdLine on Windows 2003 doesn't fail here if current directory is `C:\WINDOWS\`.

Reply:
It fails if it doesn't though.

Reply:
Correct, I will add GetWindowsDirectory/SetCurrentDirectory in the next push to make it deterministic.

Reply:
String replacement is fine, it just tends to be harder than sprintf.
You can use https://github.com/reactos/reactos/blob/c2c66aff7dacc62d125f2cd61d1167e9a2aa3fd6/modules/rostests/kmtests/ntos_io/IoFilesystem.c#L219-L225 as a basis if you like.